File: FEB25SETINTERNS1CSMSAGROUP10/FEB25SETINTERNS1CSMSAGROUP10/Infosys.EAgriculture/EAgricultureApp/src/app/EAgriculture-interfaces/farmerTransactionHistory.ts
export interface IFarmerTransactionHistory
{
  transactionID: string;
  cropName: string;
  quantityRequested: number;
  offeredPrice?: number;
  totalAmount?: number;
  requestDate?: Date;
  userID?: string;
  fullName: string;
  status: string;
}

File: FEB25SETINTERNS1CSMSAGROUP10/FEB25SETINTERNS1CSMSAGROUP10/Infosys.EAgriculture/Infosys.EAgriculture.DAL/CustomDataTransferObjectClass/FarmerTransactionHistory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Infosys.EAgriculture.DAL.CustomDataTransferObjectClass
{
    public class FarmerTransactionHistory
    {
        [Key]
        public string TransactionID { get; set; }
        public string CropName { get; set; }
        public decimal QuantityRequested { get; set; }
        public decimal? OfferedPrice { get; set; }
        public decimal? TotalAmount { get; set; }
        public DateTime? RequestDate { get; set; }
        public string? UserID { get; set; }
        public string FullName { get; set; }
        public string Status { get; set; }
    }
}
